apps/site/app/[locale]/page.tsx

```diff
@@ -150,7 +150,9 @@ const getPage: FC<DynamicParams> = async props => {
     // within a server-side context
     return (
       <MatterProvider {...sharedContext}>
-        <WithLayout layout={frontmatter.layout}>{content}</WithLayout>
+        <WithLayout layout={frontmatter.layout} modal={frontmatter.modal}>
```

Comment:
I don't like this approach of modal metadata to be injected at layout-level, it implies you cannnot have different sort of modals per page.